### PR TITLE
Add support for ssl configuration to addListener

### PR DESCRIPTION
**Pull request by cmelchior**
Add public API support controlling SSL to the global notifier.

- [x] Merge https://github.com/realm/realm-js-private/pull/510
- [x] Changelog

### CODE REVIEW

**Comment by cmelchior** [comment]
Yeah. I had the same doubt as you about the API. Creating a new function with a single config object seems like a good approach and then we can deprecate the other method.

**Comment by cmelchior** [comment]
@tgoyne This API does not appear to be used, and looking at the code I'm not sure it even works: https://github.com/realm/realm-js/blob/master/docs/sync.js#L141 Can it be safely removed?

**Comment by tgoyne** [comment]
Yes, that function is used and works. There's tests for it in realm-js-private.

File: lib/notifier.js [hunk 1]
```diff
@@ -256,10 +256,22 @@ class Listener {
...
+            // SSL Configuration not supported in the old API
+        }
+    }
+
     if (!listener) {
-        listener = new Listener(this, server, user);
+        listener = new Listener(this, config);
     }
-    return listener.add(regex, event, callback);
+    return listener.add(config.filterRegex, event, callback);
```
Comment (nirinchev) [inline lib/notifier.js]: With the new API, wouldn't `user` and `regex` effectively be `event` and `callback`?

File: lib/index.d.ts [hunk 2]
```diff
@@ -543,8 +543,25 @@ declare namespace Realm.Sync {
...
+    }
+
+    /**
+     * @deprecated, to be removed in future versions
+     */
+    function addListener(serverURL: string, adminUser: Realm.Sync.User, regex: string, name: RealmListenerEventName, changeCallback: (changeEvent: ChangeEvent) => void, sslConfiguration?: Realm.Sync.SSLConfiguration): void;
+    /**
+     * @deprecated, to be removed in future versions
+     */
+    function addListener(serverURL: string, adminUser: Realm.Sync.User, regex: string, name: RealmListenerEventName, changeCallback: (changeEvent: ChangeEvent) => Promise<void>, sslConfiguration?: Realm.Sync.SSLConfiguration): void;
```
Comment (nirinchev) [inline lib/index.d.ts]: If ssl config is not supported in the old API, should we remove it from the method signature?

File: lib/notifier.js (hunk 1 above)
Comment (cmelchior) [reply lib/notifier.js]: Yes, good catch 👍

**Pull request merged**